var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -46,68 +51,112 @@ class NetcdfC(CMakePackage, AutotoolsPackage):
+        )
+        _force_autoreconf_when.append("@4.7.3:4.7.4 +parallel-netcdf")
+
+        # See https://github.com/Unidata/netcdf-c/pull/2293
+        patch(
+            "https://github.com/Unidata/netcdf-c/commit/a7ea050ebb3c412a99cc352859d5176a9b5ef986.patch?full_index=1",
+            sha256="38d34de38bad99737d3308867071196f20a3fb39b936de7bfcfbc85eb0c7ef54",
+            when="@4.8.1",
+        )
+        _force_autoreconf_when.append("@4.8.1")
+
+    with when("@4.7.2"):
+        # Fix headers
+        # See https://github.com/Unidata/netcdf-c/pull/1505
+        patch(
+            "https://github.com/Unidata/netcdf-c/pull/1505.patch?full_index=1",
+            sha256="495b3e5beb7f074625bcec2ca76aebd339e42719e9c5ccbedbdcc4ffb81a7450",
+        )
+        # See https://github.com/Unidata/netcdf-c/pull/1508
+        patch(
+            "https://github.com/Unidata/netcdf-c/pull/1508.patch?full_index=1",
+            sha256="19e7f31b96536928621b1c29bb6d1a57bcb7aa672cea8719acf9ac934cdd2a3e",
+        )
 
     # See https://github.com/Unidata/netcdf-c/pull/2618
-    patch("4.9.0-no-mpi-yes-pnetcdf.patch", when="@4.9.0: ~mpi+parallel-netcdf")
+    patch(
+        "https://github.com/Unidata/netcdf-c/commit/00a722b253bae186bba403d0f92ff1eba719591f.patch?full_index=1",
+        sha256="25b83de1e081f020efa9e21c94c595220849f78c125ad43d8015631d453dfcb9",
+        when="@4.9.0:4.9.1~mpi+parallel-netcdf",
+    )
 
     variant("mpi", default=True, description="Enable parallel I/O for netcdf-4")
     variant("parallel-netcdf", default=False, description="Enable parallel I/O for classic files")
     variant("hdf4", default=False, description="Enable HDF4 support")
     variant("pic", default=True, description="Produce position-independent code (for shared libs)")
     variant("shared", default=True, description="Enable shared library")
     variant("dap", default=False, description="Enable DAP support")
+    variant("byterange", default=False, description="Enable byte-range I/O")
     variant("jna", default=False, description="Enable JNA support")
     variant("fsync", default=False, description="Enable fsync support")
-    variant("zstd", default=True, description="Enable ZStandard compression", when="@4.9.0:")
+    variant("nczarr_zip", default=False, description="Enable NCZarr zipfile format storage")
     variant("optimize", default=True, description="Enable -O2 for a more optimized lib")
 
-    # It's unclear if cdmremote can be enabled if '--enable-netcdf-4' is passed
-    # to the configure script. Since netcdf-4 support is mandatory we comment
-    # this variant out.
-    # variant('cdmremote', default=False,
-    #         description='Enable CDM Remote support')
+    variant("szip", default=True, description="Enable Szip compression plugin")
+    variant("blosc", default=True, description="Enable Blosc compression plugin")
+    variant("zstd", default=True, description="Enable Zstandard compression plugin")
```

Comment:
(not a request, just a comment) Here there is potential for a multi-valued variant, if it seems a good idea.

Reply:
I didn't go for a multi-valued variant because:
1. The `zstd` variant already exists and I didn't want this PR to be doomed like #32476 where I tried to touch the existing variants and it has never been accepted.
2. I couldn't come up with a good name and a set of values for a multi-valued variant. If the name is `plugins` and the possible values are `none`, `szip`, `blosc` and `zstd`, then its unclear how to interpret `plugins=none`. On one hand, that can mean "do not install plugins at all", which is possible. On the other hand, it can be "do not install szip, blosc, zstd but install the two that are supposed to be ["always available"](https://github.com/Unidata/netcdf-c/blob/b30b4e87cff202ab3c6511ade05a58019fc79796/configure.ac#L1947-L1948). However, I can put more thinking into it if you say that the first reason is irrelevant.

Reply:
Not a worry for this PR. If you do this modification in another PR[^1], ping me (repeatedly if I seem to miss the PR :slightly_smiling_face: )

[^1]: that is to say, if you think it's a modification worth doing!